core/src/test/java/nl/nn/adapterframework/filesystem/BasicFileSystemTest.java

```diff
@@ -498,4 +498,42 @@ public void basicFileSystemTestListFileShouldNotReadFolders() throws Exception {
 		
 	}
 
+	@Test
+	public void getParentOfTheDeletedFile() throws Exception {
+		String folderName = "parentFolder";
+
+		fileSystem.configure();
+		fileSystem.open();
+
+		_createFolder(folderName);
+		createFile(folderName, FILE1, "text");
+
+		F f = fileSystem.toFile(folderName, FILE1);
+
+		fileSystem.deleteFile(f);
+
+		String parentFolder = fileSystem.getParentFolder(f);
+
+		assertTrue(parentFolder.endsWith(folderName));
+	}
+
+	@Test
+	public void getParentOfTheDeletedFolder() throws Exception {
+		String folderName = "parentFolder";
+
+		fileSystem.configure();
+		fileSystem.open();
+
+		_createFolder(folderName);
+		_createFolder(folderName+"/innerFolder");
```

Comment:
This just creates a folder *named* '<folderName>/innerFolder', not a sub folder of <folderName>